src/Discord.Net.Core/Entities/Guilds/IGuild.cs

```diff
@@ -91,9 +91,9 @@ public interface IGuild : IDeletable, ISnowflakeEntity
         Task<ITextChannel> GetDefaultChannelAsync(CacheMode mode = CacheMode.AllowDownload, RequestOptions options = null);
         Task<IGuildChannel> GetEmbedChannelAsync(CacheMode mode = CacheMode.AllowDownload, RequestOptions options = null);
         /// <summary> Creates a new text channel. </summary>
-        Task<ITextChannel> CreateTextChannelAsync(string name, RequestOptions options = null);
+        Task<ITextChannel> CreateTextChannelAsync(string name, RequestOptions options = null, Action<TextChannelProperties> func = null);
```

Comment:
I really don't think having the action being the last parameter is worth avoiding making a breaking change. Just way too inconsistent.